zkevm-circuits/src/evm_circuit/execution/error_oog_constant.rs

```diff
@@ -144,10 +207,124 @@ mod test {
     }
 
     #[test]
-    fn oog_constant_gadget_simple() {
-        // Transfer 1 ether, successfully
-        // in root call
+    fn test_oog_constant_root() {
+        // in root call , out of gas constant happens
         test_oog_constant(mock_tx(eth(1), gwei(2), vec![]), false);
-        // TODO: add internal call test
+    }
+
+    #[derive(Clone, Copy, Debug, Default)]
+    struct Stack {
+        gas: u64,
+        value: Word,
+        cd_offset: u64,
+        cd_length: u64,
+        rd_offset: u64,
+        rd_length: u64,
+    }
+
+    fn caller() -> Account {
+        let terminator = OpcodeId::REVERT;
+
+        let stack = Stack {
+            gas: 200,
```

Comment:
If we change this to `gas: 10`, then the test `test_oog_constant_internal` will fail, since we restore the wrong gas for caller.

Reply:
yes, after fix above comment, this is gone now. thanks for catching !